module/x/peggy/types/pool.go

```diff
@@ -85,7 +85,14 @@ func NewVoucherDenom(tokenContractAddr EthereumAddress, erc20Symbol string) Vouc
 	denomTrace := fmt.Sprintf("%s/%s/", tokenContractAddr.String(), erc20Symbol)
 	var hash tmbytes.HexBytes = tmhash.Sum([]byte(denomTrace))
 	simpleVoucherDenom := VoucherDenomPrefix + DenomSeparator + hash.String()
-	sdkVersionHackDenom := strings.ToLower(simpleVoucherDenom[0:15]) // todo: up to 15 chars (lowercase) allowed in this sdk version only
+	var symbolSliceLen int
+	// handle erc20symbol values that are shorter than 4 char
+	if len(erc20Symbol) > 4 {
+		symbolSliceLen = 4
+	} else {
+		symbolSliceLen = len(erc20Symbol)
+	}
+	sdkVersionHackDenom := strings.ToLower(simpleVoucherDenom[0:11] + erc20Symbol[0:symbolSliceLen]) // todo: up to 15 chars (lowercase) allowed in this sdk version only
```

Comment:
I was following the IBC design to have conflict free denominators. Adding the erc20 symbol into the denum may be more readable with 1 ERC20 token type but as soon as we have a second with the same symbol it is not clear anymore and can be very confusing for the users. A wallet or any other client should do a lookup for the token denominator instead to receive all the meta data like symbol, contract address, decimals,....
The logic is already implemented for the CLI no REST endpoint though.
If duplicate symbols are not possible by design then we do not need the hash.

Reply:
if we have two of the same symbol you'll have to check the hash. I don't understand how that's any worse than checking peggy<hash> value in the first place. Unless you figure wallets will do it in an automated manner?

But even if we do have wallets perform the lookup, the hash is the same length and the format is still machine parseable. Please outline how this is not a strict improvement?

Reply:
I was expecting that the symbols are not globally unique. For example with multiple BTC symbols from different ERC20 contracts, how can a user pick the "right" one? This would open the door for scams.
If conflicting symbols are prevented by design (as the ERC20 contracts are whitelisted on the bridge) then we can just make it the `peggy/btc` voucher. No need for the hash...
Personally I would prefer the IBC approach and not encode symbols into the denom. We would provide the ERC20 contract address, symbol and other metadata for the wallets/users by an endpoint.